### DISCUSSION TITLE
CreateIssuedDocumentRequest

### DISCUSSION THREAD

**Discussion by mastsolutions** [fattureincloud/api#197] [Q&A] [answered]
Buongiorno,

vi chiedo un aiuto per capire come passare una lista di articoli presenti in un Dataset nella parte evidenziata di codice
Ne ho provate di tutte, grazie 1000   

Grazie 1000
Stefano

       IssuedDocument quote = new IssuedDocument(
                type: IssuedDocumentType.Quote,
                entity: entity,
                date: _documentType.Item1,
                number: _documentType.Item2,
                numeration: _documentType.Item3,
                //subject: "internal sub, TEST FIC API V2",
                visibleSubject: _documentType.Item4,
                currency: new Currency(id: "EUR"),
                language: new Language(code: "it", name: "italiano"),
                itemsList: new List<IssuedDocumentItemsListItem>
                {
                    **new IssuedDocumentItemsListItem(productId: 22, code: "TV220", name: "Tavolo in legno", category: "HW", description: "mogano laccato", netPrice: 100, discount: 20, discountHighlight: true, qty: 1, vat: new VatType(id: 0)),
                    new IssuedDocumentItemsListItem(productId: 101, code: "TV3", name: "Tavolo in legno", category: "HW", description: "mogano laccato", netPrice: 100, discount: 20, discountHighlight: true, qty: 1, vat: new VatType(id: 0)),**

                }
            );

**Comment by valmoz** [comment, marked as answer]
Buongiorno,
da ciò che vedo, il suo codice mi sembra essere Java.

Il campo itemsList è una istanza di _List<IssuedDocumentItemsListItem>_; quindi dovrà instanziare degli oggetti di tipo _IssuedDocumentItemsListItem_, uno per articolo, e inserirli in un oggetto che rispetti l'interfaccia List, come ad esempio ArrayList.

Per creare una lista può leggere [questa guida](https://www.geeksforgeeks.org/initializing-a-list-in-java/).

Un esempio di codice potrebbe essere il seguente:
```java
List<IssuedDocumentItemsListItem> itemsList = new ArrayList<IssuedDocumentItemsListItem>();
 itemsList.add(new IssuedDocumentItemsListItem(productId: 22, code: "TV220", name: "Tavolo in legno", category: "HW", description: "mogano laccato", netPrice: 100, discount: 20, discountHighlight: true, qty: 1, vat: new VatType(id: 0)));
 itemsList.add(new IssuedDocumentItemsListItem(productId: 101, code: "TV3", name: "Tavolo in legno", category: "HW", description: "mogano laccato", netPrice: 100, discount: 20, discountHighlight: true, qty: 1, vat: new VatType(id: 0)));

...

// all'interno dell'oggetto IssuedDocument
itemsList: itemsList,
```

Restiamo a disposizione per ulteriori domande.

Grazie
Mauro

**Comment by mastsolutions** [comment]
Buongiorno, 
grazie per la risposta, il codice è C#
Provo comunque a seguire il link che mi ha inviato
Stefano

**Reply by valmoz** [reply]
In tal caso, può provare in maniera del tutto simile con [questo link](https://www.c-sharpcorner.com/UploadFile/mahesh/create-a-list-in-C-Sharp/).

Grazie

**Comment by mastsolutions** [comment]
Buonasera Mauro,
funziona tutto perfettamente, questo il codice messo a punto

```cs
List<IssuedDocumentItemsListItem> itemsList = new ArrayList<IssuedDocumentItemsListItem>();
            for (int x = 0; x<_items.Rows.Count; x++)
            {
                itemsList.Add(new IssuedDocumentItemsListItem(productId: (int)_items.Rows[x].ItemArray[0], code: (string)_items.Rows[x].ItemArray[1], .... );
            }

...

            IssuedDocument quote = new IssuedDocument(
                type: IssuedDocumentType.Quote,
                entity: entity,
                date: _documentType.Item1,
                number: _documentType.Item2,
                numeration: _documentType.Item3,
                //subject: "internal sub, TEST FIC API V2",
                visibleSubject: _documentType.Item4,
                currency: new Currency(id: "EUR"),
                language: new Language(code: "it", name: "italiano"),
                itemsList: itemsList
            );
```
1000 Grazie
Stefano

**Reply by valmoz** [reply]
Ottimo, buona serata

Mauro
